### PR TITLE
Support Capture and Authorize in Demo App

### PR DESCRIPTION
**Pull request by sshropshire**
### Summary of changes

- In demo app, either capture or authorize an order based on the original intent
- Related to: https://github.com/paypal/Android-SDK/pull/143

### Checklist

~- [ ] Added a changelog entry~

- @sshropshire
- @KunJeongPark

### CODE REVIEW

File: Demo/Demo/Models/EmptyBodyParams.swift [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import Foundation
+
+struct EmptyBodyParams: Codable {}
```
Comment (sshropshire) [inline Demo/Demo/Models/EmptyBodyParams.swift:3]: We needed to create this to make an empty `POST` request for Order capture and authorize calls.

**Review by jcnoriega — APPROVED** [review]

File: Demo/Demo/ViewModels/BaseViewModel.swift [hunk 2]
```diff
@@ -178,9 +178,32 @@ class BaseViewModel: ObservableObject, PayPalWebCheckoutDelegate, CardDelegate {
     // MARK: - Card Delegate
 
     func card(_ cardClient: CardClient, didFinishWithResult result: CardResult) {
-        updateTitle("Order Id:\(result.orderID) status: \(result.status)\n \(String(describing: result.paymentSource))")
+        let intent = DemoSettings.intent.rawValue.uppercased()
+        if intent == "CAPTURE" {
+            updateTitle("Capturing Order with Id:\(result.orderID)...")
```
Comment (jaxdesmarais) [inline Demo/Demo/ViewModels/BaseViewModel.swift]: nit as iOS prefers `ID` over `Id`: 
```suggestion
            updateTitle("Capturing Order with ID:\(result.orderID)...")
```

File: Demo/Demo/ViewModels/BaseViewModel.swift [hunk 3]
```diff
@@ -178,9 +178,32 @@ class BaseViewModel: ObservableObject, PayPalWebCheckoutDelegate, CardDelegate {
...
+            captureOrderOnMerchantServer(result: result)
+        } else if intent == "AUTHORIZE" {
+            updateTitle("Authorizing Order with Id:\(result.orderID)...")
```
Comment (jaxdesmarais) [inline Demo/Demo/ViewModels/BaseViewModel.swift]:
```suggestion
            updateTitle("Authorizing Order with ID:\(result.orderID)...")
```

File: Demo/Demo/ViewModels/BaseViewModel.swift [hunk 4]
```diff
@@ -178,9 +178,32 @@ class BaseViewModel: ObservableObject, PayPalWebCheckoutDelegate, CardDelegate {
...
+            authorizeOrderOnMerchantServer(result: result)
+        }
     }
-
+    
+    private func captureOrderOnMerchantServer(result: CardResult) {
+        Task {
+            let captureResult = try? await DemoMerchantAPI.sharedService.caputureOrder(orderID: result.orderID)
+            let status = captureResult?.status ?? result.status
+            updateTitle("Order Id:\(result.orderID) status: \(status)\n \(String(describing: result.paymentSource))")
```
Comment (jaxdesmarais) [inline Demo/Demo/ViewModels/BaseViewModel.swift]:
```suggestion
            updateTitle("Order ID:\(result.orderID) status: \(status)\n \(String(describing: result.paymentSource))")
```

File: Demo/Demo/ViewModels/BaseViewModel.swift [hunk 5]
```diff
@@ -178,9 +178,32 @@ class BaseViewModel: ObservableObject, PayPalWebCheckoutDelegate, CardDelegate {
...
+        }
+    }
+    
+    private func authorizeOrderOnMerchantServer(result: CardResult) {
+        Task {
+            let authorizeResult = try? await DemoMerchantAPI.sharedService.authorizeOrder(orderID: result.orderID)
+            let status = authorizeResult?.status ?? result.status
+            updateTitle("Order Id:\(result.orderID) status: \(status)\n \(String(describing: result.paymentSource))")
```
Comment (jaxdesmarais) [inline Demo/Demo/ViewModels/BaseViewModel.swift]:
```suggestion
            updateTitle("Order ID:\(result.orderID) status: \(status)\n \(String(describing: result.paymentSource))")
```

**Review by jaxdesmarais — APPROVED** [review]

**Pull request merged**